pages/api/mechanic/getSingleMechanic.ts

```diff
@@ -12,19 +12,15 @@ export default async function handler(
     try {
       // Call the authorize middleware to extract user ID from token
       authorize(req, res, async () => {
-        const user = req["user"]; 
-        const userId = user.id;
+        const user = req["user"];
 
         // Validate user ID
-        if (!userId) {
+        if (!user?.id) {
           return res.status(400).json({ message: "User ID is required" });
         }
 
-        // Find mechanic by user ID
-        const mechanic = await MechanicRegistrationModel.findOne({
-          user: userId,
-        });
-
+        // Find mechanic by ID
+        const mechanic = await MechanicRegistrationModel.findById(user.id);
```

Comment:
Observed the file uses findById method for this purpose, so changed this to maintain the consistency in the file.